Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +15,141 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int FindDigit(string equation)
+        {
+            string[] terms = equation.Split(new Char[] { '*', '=' });
+            /*The first operand has a missing digit*/
+            if (terms[0].Contains('?'))
+            {
+                int b = Int32.Parse(terms[1]);
+                int c = Int32.Parse(terms[2]);
+                if (c % b != 0)
+                {
+                    return (-1);
+                }
+                int a = c / b;
+                string s = a.ToString();
+                int indexofq = 0;
+                int matches = 0;
+                string A = terms[0];
+                if (s.Length == A.Length)
+                {
+                    for (int i = 0; i < s.Length; i++)
+                    {
+                        if (A[i] == '?')

```

Comment:
Always check string comparison with string equals.
Also please read difference between == and Equals

Reply:
Used Equals() function.